### PR TITLE
Prototype Design pattern : Example code for Shape Prototype System

### PR DESCRIPTION
**Pull request by JotsnaW**
Created Prototype design pattern Code

### CODE REVIEW

File: PrototypeDesignPattern1/Circle.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace PrototypeDesignPattern1
+{
+    internal class Circle: IComponent
+    {
+        public double Radius {  get; set; }
```
Comment (asadhasnain) [inline PrototypeDesignPattern1/Circle.cs]: You are accepting the value in constructor than its better to mark it as private

File: PrototypeDesignPattern1/Rectangle.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            m_length = length;
+            m_width = width;
+            m_area = m_length * m_length;
+        }
+
+        // clone method
+        public IComponent Clone()
+        {
+            Rectangle clonedRectangle = new Rectangle(m_length, m_width);
```
Comment (asadhasnain) [inline PrototypeDesignPattern1/Rectangle.cs]: Update it, for example Rectangle is a collection of lines, and you can try by cloning the lines too.

**Review by asadhasnain — CHANGES_REQUESTED** [review]

File: PrototypeDesignPattern1/Rectangle.cs (hunk 2 above)
Comment (JotsnaW) [reply PrototypeDesignPattern1/Rectangle.cs]: Updated code

File: PrototypeDesignPattern1/Circle.cs (hunk 1 above)
Comment (JotsnaW) [reply PrototypeDesignPattern1/Circle.cs]: Updated code

**Pull request merged**